### PR TITLE
[@kadena/graph] , [@kadena/graph-client] Implement crossChainTransfer for Transfers and display them as one record (Feat)

### PR DESCRIPTION
**Pull request by nil-amrutlal-dept**
image
image

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205801361945267

### CODE REVIEW

File: packages/apps/graph/src/graph/Query/transfers.ts [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+      accountName: t.arg.string({ required: false }),
+      moduleName: t.arg.string({ required: false }),
       chainId: t.arg.string({ required: false }),
     },
     type: 'Transfer',
     cursor: 'blockHash_chainId_orderIndex_moduleHash_requestKey',
+    async totalCount(__parent, args) {
+      try {
+        return await prismaClient.transfer.count({
+          where: geerateTransferFilter(args),
```
Comment (MRVDH) [inline packages/apps/graph/src/graph/Query/transfers.ts:18]: typo in function name

File: packages/apps/graph/src/graph/objects/Transfer.ts [hunk 2]
```diff
@@ -17,6 +17,68 @@ export default builder.prismaNode('Transfer', {
...
+          return await prismaClient.transfer.findFirst({
+            where: {
+              blockHash: counterTransaction.blockHash,
+              requestKey: counterTransaction.requestKey,
+              OR: [{ senderAccount: '' }, { receiverAccount: '' }],
+              amount: parent.amount,
+            },
+          });
+        } catch (error) {
+          throw normalizeError(error);
```
Comment (MRVDH) [inline packages/apps/graph/src/graph/objects/Transfer.ts:77]: Did you play around with the input params to see if you can generate errors that are not yet covered in the 'normalizeError' function? Or vague errors that could use a 'description' field in the error object.

Comment (nil-amrutlal-dept) [reply packages/apps/graph/src/graph/objects/Transfer.ts:77]: Yep the only error that could potentially occur is not finding the corresponding initiating transfer and that one logs correctly on the client. And for me to replicate this error I had to manually change data in the database. I dont think there are a lot of possible errors 

image

**Review by MRVDH — APPROVED** [review]

File: packages/apps/graph-client/src/components/compact-transfers-table/compact-transfers-table.tsx [hunk 3]
```diff
@@ -56,10 +56,24 @@ export const CompactTransfersTable = (
         </Table.Head>
         <Table.Body>
           {transfers.edges.map((edge, index) => {
+            /**  These transfers are going to be added to their crosschain counterpart and
+             this way we avoid repeated transfers in the table */
+            if (!chainId && edge?.node.transaction?.pactId) {
+              return <></>;
+            }
```
Comment (alber70g) [inline packages/apps/graph-client/src/components/compact-transfers-table/compact-transfers-table.tsx:63]: To convey why this is, we could move the check to a function with a descriptive name.
Something like `isCrosschainCounterpart(edge)`?

Comment (nil-amrutlal-dept) [reply packages/apps/graph-client/src/components/compact-transfers-table/compact-transfers-table.tsx:63]: Sounds good! I am currently working on a PR to include item resizing and some other aspects related to the transfers page. I could add this there 😃

File: packages/apps/graph/generated-schema.graphql [hunk 4]
```diff
@@ -180,9 +180,9 @@ type Query {
   nodes(ids: [ID!]!): [Node]!
   pactQueries(pactQuery: [PactQuery!]!): [String!]!
   pactQuery(pactQuery: PactQuery!): String!
-  transactions(accountName: String, after: String, before: String, blockHash: String, chainId: String, first: Int, last: Int, moduleName: String): QueryTransactionsConnection!
+  transactions(accountName: String, after: String, before: String, blockHash: String, chainId: String, first: Int, last: Int, moduleName: String, requestKey: String): QueryTransactionsConnection!
```
Comment (alber70g) [inline packages/apps/graph/generated-schema.graphql:183]: Is `requestKey` a remnant of debugging?

Comment (nil-amrutlal-dept) [reply packages/apps/graph/generated-schema.graphql:183]: Yes indeed. I am using it for debugging purposes I can remove it

**Pull request merged**